src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -349,19 +381,37 @@ def test_omp_do_directive_validate_global_constraints(fortran_reader,
 
     directive = tree.walk(OMPParallelDoDirective)
 
-    # The first loop nest will fail the validation
+    # The first and second loop nests will fail the validation
     with pytest.raises(GenerationError) as err:
         _ = fortran_writer(directive[0])
-    assert ("OMPParallelDoDirective must have as many immediately nested loops"
-            " as the collapse clause specifies but 'OMPParallelDoDirective"
-            "[omp_schedule=auto,collapse=2]' has a collapse=2 and the nested "
-            "statement at depth 1 is a Assignment rather than a Loop."
+    assert ("OMPParallelDoDirective must have as many immediately nested "
```

Comment:
If the two messages are identical then you could do:

    for dirv in directive[0:2]:
       ...

Reply:
Done

Reply:
Close! You needed [0:2] rather than [0:1] because of Python's weird slice notation. I've taken the liberty of verifying that this correctly gives two trips round the loop and have made the change.